modules/beta-public-cluster/variables.tf

```diff
@@ -350,9 +350,10 @@ variable "enable_vertical_pod_autoscaling" {
   default     = false
 }
 
-variable "identity_namespace" {
-  type        = string
-  description = "Workload Identity namespace"
-  default     = ""
+variable "workload_identity_config" {
+  description = "Workload Identity configuration"
+  type        = list(object({ identity_namespace = string }))
+  default     = []
```

Comment:
We shouldn't force users to specify this as a list. I'd like to keep the API the same.

We should be able to build a list within the module based on the value provided.

Reply:
Done, The module builds the list now and uses former API .
I ran it here without identity_namespace and the cluster has been created successfully.